### ISSUE TITLE
Add user-generated warning macro

### ISSUE THREAD

**Issue by hydrogen-mvm** [alliedmodders/sourcemod#918]
Often times I accidentally compile plugins with the `#define DEBUG` (or some debug flag) enabled, and then I ship them to production because they worked in the testing environment, only for it to hilariously break in the live environment. For example, it sends data to localhost instead of a real database server.

Something like this would be really handy to warn that a plugin was compiled with debug mode enabled:

```
#if defined DEBUG
	#warning This plugin is compiled in debug mode!
#endif
```

In the compiler console it would print something like "warning: user-generated warning: This plugin is compiled in debug mode!" but the compile would still continue as it's just a warning. It would serve as a friendly nudge to the developer "hey reminder: don't compile with that flag on when releasing it!"

Naturally when shipping something live you want the compiler to emit 0 warnings and 0 errors, so for most people that feature should be good enough to aid their development.

There exists such a macro for error but, as the name implies, this aborts the compilation, so it's not appropriate for this use.

```
#if !defined CC_MAIN
	#error This file must be compiled from mvm_canteen_crasher.sp
#endif
```

Thanks.

**Comment by Headline** [comment]
Thanks for creating this issue! Luckly for you this was already merged in sourcepawn! alliedmodders/sourcepawn#170

It should be currently active in the 1.10 branch of sourcemod

**Comment by hydrogen-mvm** [comment]
Oh I see, sorry I use the stable builds (1.9 as of writing) since I'm not sure how stable the dev builds are.

Thanks for letting me know!
